src/exe/Main.hs

```diff
@@ -125,8 +134,29 @@ optionDescriptors =
     (  "Optional. Path to output directory.\n"
     ++ "Prints to the console by default."
     )
+  , Option
+    ['f']
+    ["frontend"]
+    (ReqArg (\f opts -> opts { optFrontend = f }) "FRONTEND")
+    (  "Optional. Specifies the front end for the compiler to use.\n"
+    ++ "Allowed values are: `ghc-lib`, `haskell-src-extensions`.\n"
+    ++ "Uses haskell-src-extensions by default."
+    )
   ]
 
+-- | Parses a given string to one of the frontents.
+parseFrontend :: Member Report r => String -> Sem r Frontend
+parseFrontend s = case Map.lookup s frontendMap of
+  Nothing ->
+    reportFatal
+      $  Message Error
+      $  "Unavailable frontend.\n"
+      ++ "Use '--help' for allowed values."
+  Just f -> return f
+ where
+  frontendMap =
+    Map.fromList [("haskell-src-extensions", HSE), ("ghc-lib", GHClib)]
```

Comment:
I think that we should allow abbreviated spellings, because they shorten the length of the command (especially with `haskell-src-exts`, where we can save 13 characters). I don't see why they should be confusing for the user, they are very similar to the one-letter option descriptors after all, so one abbreviation per front end should be fine:
```suggestion
    Map.fromList [("haskell-src-extensions", HSE), ("hse", HSE), ("ghc-lib", GHClib), ("ghc", GHCLib)]
```